src/components/media/OutputsGallery.tsx

```
@@ -563,19 +676,22 @@ export const OutputsGallery: React.FC<OutputsGalleryProps> = ({
-      const currentPathFilesKeys = folderContent.files.map(f => `${f.filename}-${f.subfolder}-${f.type}`);
-      const allCurrentFilesSelected = currentPathFilesKeys.every(key => selectedFiles.has(key));
+      // In folder mode, select currently rendered files by default.
+      const selectableFiles = visibleOnly ? visibleFiles : folderContent.files;
+      const currentPathFilesKeys = selectableFiles.map(f => `${f.filename}-${f.subfolder}-${f.type}`);
+      const currentPathFolderKeys = folderContent.folders.map(folder => `folder:${folder.fullPath}`);
+      const currentPathSelectableKeys = [...currentPathFilesKeys, ...currentPathFolderKeys];
+      const allCurrentFilesSelected = currentPathSelectableKeys.length > 0 && currentPathSelectableKeys.every(key => selectedFiles.has(key));
 
       if (allCurrentFilesSelected) {
-        currentPathFilesKeys.forEach(key => newSelected.delete(key));
+        currentPathSelectableKeys.forEach(key => newSelected.delete(key));
```

Biome (2.4.6)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)